elements/lisk-cryptography/src/ed.ts

```diff
@@ -99,10 +82,15 @@ export interface SignedMessageWithPrivateKey {
 export const signMessageWithPrivateKey = (
 	message: string,
 	privateKey: Buffer,
+	tag = MESSAGE_TAG_NON_PROTOCOL_MESSAGE,
```

Comment:
I forgot to mention, i think we should make it possible to have message as `Buffer`. (considering we want to use this to sign register multi-signature object)
to keep the compatibility, we could do like
```
message: string | Buffer,
```
and
```
const msgBytes = typeof message === 'string' ? Buffer.from(message, 'utf8') : message;
```

Reply:
The function currently returns an object which includes a `message` property. If we allow `message` to be both `string` and `Buffer`, should we then go ahead with [what I've proposed here](https://github.com/LiskHQ/lisk-sdk/pull/8805#discussion_r1286513995) and have the function return only `signature`?

Reply:
ah i see. maybe we can return what was given not to make breaking change?